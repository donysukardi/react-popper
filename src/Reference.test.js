// @flow
import React from 'react';
import warning from 'warning';
import { mount } from 'enzyme';

// Public API
import { Reference } from '.';

// Private API
import { ManagerContext } from './Manager';

jest.mock('warning');

describe('Arrow component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders the expected markup', () => {
    const setReferenceNode = jest.fn();

    // HACK: wrapping DIV needed to make Enzyme happy for now
    const wrapper = mount(
      <ManagerContext.Provider
        value={{
          setReferenceNode,
          referenceNode: undefined,
        }}
      >
        <Reference>{({ ref }) => <div ref={ref} />}</Reference>
      </ManagerContext.Provider>
    );
    expect(wrapper.children()).toMatchSnapshot();
  });

  it('consumes the ManagerContext from Manager', () => {
    const setReferenceNode = jest.fn();

    // HACK: wrapping DIV needed to make Enzyme happy for now
    mount(
      <div>
        <ManagerContext.Provider
          value={{
            setReferenceNode,
            referenceNode: undefined,
          }}
        >
          <Reference>{({ ref }) => <div ref={ref} />}</Reference>
        </ManagerContext.Provider>
      </div>
    );
    expect(setReferenceNode).toHaveBeenCalled();
  });

  it('warns when setReferenceNode is present', () => {
    const setReferenceNode = jest.fn();

    // HACK: wrapping DIV needed to make Enzyme happy for now
    mount(
      <div>
        <ManagerContext.Provider
          value={{
            setReferenceNode,
            referenceNode: undefined,
          }}
        >
          <Reference>{({ ref }) => <div ref={ref} />}</Reference>
        </ManagerContext.Provider>
      </div>
    );
    expect(warning).toHaveBeenCalledWith(
      true,
      '`Reference` should not be used outside of a `Manager` component.'
    );
  });

  it('does not warn when setReferenceNode is not present', () => {
    // HACK: wrapping DIV needed to make Enzyme happy for now
    mount(
      <div>
        <ManagerContext.Provider
          value={{
            referenceNode: undefined,
          }}
        >
          <Reference>{({ ref }) => <div ref={ref} />}</Reference>
        </ManagerContext.Provider>
      </div>
    );
    expect(warning).toHaveBeenCalledWith(
      false,
      '`Reference` should not be used outside of a `Manager` component.'
    );
  });
});
