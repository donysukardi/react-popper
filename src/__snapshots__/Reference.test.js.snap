// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow component renders the expected markup 1`] = `
<InnerReference
  setReferenceNode={
    [MockFunction] {
      "calls": Array [
        Array [
          <div />,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div />
</InnerReference>
`;
